src/basic/chattr-util.c

```diff
@@ -172,3 +173,93 @@ int read_attr_at(int dir_fd, const char *path, unsigned *ret) {
 
         return read_attr_fd(fd, ret);
 }
+
+int read_xattr_flags_fd(int fd, unsigned *ret) {
+        struct fsxattr attrs;
+        _cleanup_close_ int fd_reopened = -EBADF;
+
+        assert(ret);
+
+        fd = fd_reopen_condition(fd, 0, O_PATH|O_CLOEXEC|O_NOCTTY|O_RDONLY, &fd_reopened);
+        if (fd < 0)
+                return fd;
+
+        if (ioctl(fd, FS_IOC_FSGETXATTR, &attrs) < 0)
+                return -errno;
+
+        *ret = attrs.fsx_xflags;
+
+        return 0;
+}
+
+int get_proj_id(int fd, uint32_t *ret) {
+        struct fsxattr attrs;
+        _cleanup_close_ int fd_reopened = -EBADF;
+
+        assert(ret);
+
+        fd = fd_reopen_condition(fd, 0, O_PATH|O_CLOEXEC|O_NOCTTY|O_RDONLY, &fd_reopened);
```

Comment:
same here.
- missing assertion for `fd`,
- spurious arguments for fd_reopen_condition().